### PR TITLE
feat(error): allow to present some passing examples on failure

### LINKED ISSUE #93: feat(error): allow to present some passing examples on failure
**Is your feature request related to a problem? Please describe.**
When the action fails, the log shows the regex that's been used, but a regex is not super easy to read.

**Describe the solution you'd like**
I'd love to add an optional param to display some matching examples.

Example of better log:
```
Error: Pull Request title "(PIX-0000) Linter les titres de PR" failed to pass match regex - /\[[A-Z]+\] .+/

Example of matching titles:
- "[FEATURE] Migration des profils cibles (PIX-6596)"
- "[BUGFIX] Corriger l'erreur d'année sur un test CPF (pix-6704)"
- "[TECH] Linter les titres de PR"
```

Another approach would be to redirect the user to the full rules (e.g.: a CONTRIBUTING.md file).

**Describe alternatives you've considered**
It should be possible to do it on my side by adding a "failure" step that catches the error and logs it itself.

**Additional context**
I've deployed this action in the [Pix project](https://github.com/1024pix/pix/pull/5445) :)

### PR DESCRIPTION
**Pull request by Slashgear**
## 🤔 Why ?

Fix #93

## 💻 How ?

- add documentation
- update action.yml definition
- implement it in run.ts
- test it with Jest

### CODE REVIEW

File: src/run.ts [hunk 1]
```diff
@@ -15,9 +15,16 @@ export const run = (context: Context) => {
   info(`Pull Request title: "${pullRequestTitle}"`);
 
   const regex = RegExp(getInput("regexp"));
+  const helpMessage = getInput("helpMessage");
```
Comment (yannbertrand) [inline src/run.ts:18]: Didn't know about `"@actions/core"`, thanks for the discovery 🙏

File: src/run.ts [hunk 2]
```diff
@@ -15,9 +15,16 @@ export const run = (context: Context) => {
...
   if (!regex.test(pullRequestTitle)) {
-    setFailed(
-      `Pull Request title "${pullRequestTitle}" failed to pass match regex - ${regex}`
-    );
+    let message = `Pull Request title "${pullRequestTitle}" failed to pass match regexp
+Used Regexp: ${regex}
```
Comment (yannbertrand) [inline src/run.ts:21]: Not sure the new line clarifies the log, I think I preferred it before. Do you think it's easier to read this way?

**Review by yannbertrand — APPROVED** [review]
Looking good 👍

File: src/run.ts (hunk 2 above)
Comment (Slashgear) [reply src/run.ts:21]: As you prefer :)

**Pull request merged**